cmake/settings.cmake

```diff
@@ -1,14 +1,56 @@
 # Copyright (c) Microsoft Corporation
 # SPDX-License-Identifier: MIT
 
+# Some projects do not play well with the default definitions (such
+# as bpftool and -DWIN32). Remove them from the variables for now
+set(settings_variable_list
+  "CMAKE_C_FLAGS_RELEASE"
+  "CMAKE_C_FLAGS_DEBUG"
+  "CMAKE_C_FLAGS_RELWITHDEBINFO"
+
+  "CMAKE_CXX_FLAGS_RELEASE"
+  "CMAKE_CXX_FLAGS_DEBUG"
+  "CMAKE_CXX_FLAGS_RELWITHDEBINFO"
+
+  "CMAKE_C_FLAGS"
+  "CMAKE_CXX_FLAGS"
+)
+
+foreach(settings_variable ${settings_variable_list})
+  string(REPLACE "/D_WINDOWS" "" "${settings_variable}" ${${settings_variable}})
+  string(REPLACE "/DWIN32" "" "${settings_variable}" ${${settings_variable}})
+endforeach()
+
+add_library("ebpf_for_windows_base_definitions" INTERFACE)
+if(CMAKE_GENERATOR MATCHES "Visual Studio")
+  set(base_definition_list
+    $<$<CONFIG:Debug>:_DEBUG>
+    $<$<CONFIG:Release>:NDEBUG>
+  )
```

Comment:
FYI the ebpf-verifier project seems to have no problem getting NDEBUG to appear in the PreprocessorDefinitions in the generated vcxproj file without any cmake/settings.cmake or mention in the the CMakeLists.txt file.